dlrover/python/master/hyperparams/simple_strategy_generator.py

```diff
@@ -0,0 +1,163 @@
+from dlrover.python.master.hyperparams.strategy_generator import (
+    StrategyGenerator,
+)
+from dlrover.python.master.stats.reporter import JobMeta, LocalStatsReporter
+
+gpu_stats = [
+    {
+        "index": 0,
+        "total_memory_gb": 24,
+        "used_memory_gb": 0,
+    },
+    {
+        "index": 1,
+        "total_memory_gb": 24,
+        "used_memory_gb": 2.631,
+    },
+]
+
+# Define model configuration as a dictionary
+mock_model_config = {
+    "block_size": 128,
+    "n_layer": 6,
+    "n_heads": 6,
+    "n_embd": 384,
+}
+
+mock_dataloader_config = DataLoaderConfig(0, "simple_dataloader", 32, 2, 0)
+
+
+class SimpleStrategyGenerator(StrategyGenerator):
+    """
+    A simple strategy generator for local optimization.
+    This class is responsible for generating optimal configurations for data
+    loaders and optimizers
+    """
+
+    def __init__(self, job_uuid):
+        self._job_uuid = job_uuid
+        self._stats_collector = LocalStatsReporter(JobMeta(job_uuid))
+
+    def start(self):
+        try:
+            logger.info("Start local strategy generator")
+            threading.Thread(
+                target=self._extract_node_resource,
+                name="strategy-generator",
+                daemon=True,
+            ).start()
+        except Exception as e:
+            logger.error(e)
+
+    def generate_opt_strategy(
+        self,
+        gpu_stats=[],
+        model_config=mock_model_config,
+    ) -> ParallelConfig:
+        node_samples = self._extract_node_resource()
+        paral_configs: Dict[str, ParallelConfig] = {}
+        for nodes in node_samples[NodeType.WORKER]:
+            for node in nodes:
+                gpu_stats = node.used_resource.gpu_stats
+                paral_config = node.paral_config
+                data_loader_config, optimizer_config = (
+                    self._generate_dataloader_config(
+                        gpu_stats, model_config, paral_config.dataloader
+                    ),
+                    self._generate_optimizer_config(),
+                )
+                paral_configs[node.name] = ParallelConfig(
+                    data_loader_config, optimizer_config
+                )
+        logger.info("paral configs: %s", paral_configs["simple_node"])
+        return paral_configs["simple_node"]
+
+    def _generate_dataloader_config(
+        self, gpu_stats, model_config, dataloader_config
+    ):
+        if gpu_stats == []:
+            return DataLoaderConfig(0, "", 0, 0, 0)
+        # Calculate the minimum remaining memory among GPUs
+        min_remain_memory = min(
+            entry["total_memory_gb"] - entry["used_memory_gb"]
+            for entry in gpu_stats
+        )
+
+        # Update dataloader configuration version
+        updated_version = dataloader_config.version + 1
+        # Extract dataloader configuration values
+        batch_size = dataloader_config.batch_size
+        logger.info("batch size: %d", batch_size)
```

Comment:
We can `logger.info(f"batch size: {batch_size}")`

Reply:
Fixed by comment.